### PR TITLE
version bump

### PR DESCRIPTION
**Pull request by saif-ellafi**
Updates Scala to `2.11.12`, Spark to `2.4.0` and Zeppelin to `0.8.1`

### CODE REVIEW

File: cluster-setup/local-cluster/docker-compose.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ services:
       - "8020:8020"
       - "8088:8088"
       - "9001:9001"
-      - "50070:50070"
-      - "50090:50090"
+      - "50570:50570"
```
Comment (robvadai) [inline cluster-setup/local-cluster/docker-compose.yml]: Why did you change the port mapping?

File: cluster-setup/local-cluster/docker-compose.yml [hunk 2]
```diff
@@ -43,3 +43,24 @@ services:
     links:
       - hadoop-namenode
     tty: true
+  hadoop-datanode2:
```
Comment (robvadai) [inline cluster-setup/local-cluster/docker-compose.yml]: My idea was that we have 1 datanode by default. You can of course add more but I thought by default we have 1, to keep it simple.

File: docker-container/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM isuper/java-oracle:jdk_8
+FROM larskghf/oracle-java-jdk8
```
Comment (robvadai) [inline docker-container/Dockerfile]: Why did you change the Docker base image?

**Comment by robvadai** [comment]
Thank you for the contribution @saifjsl !
I have a few comments, can you look at them please before I merge this PR.

**Comment by robvadai** [comment]
Oh, and can you update the README as well please with the updated versions.

File: cluster-setup/local-cluster/docker-compose.yml (hunk 1 above)
Comment (saifjsl) [reply cluster-setup/local-cluster/docker-compose.yml]: Hi, mostly because I wanted to try this from windows, and the original range is protected by Windows. I couldn't really make everything work on Windows but that was a safe change that could make someone's life easier.

File: cluster-setup/local-cluster/docker-compose.yml (hunk 2 above)
Comment (saifjsl) [reply cluster-setup/local-cluster/docker-compose.yml]: I had the feeling two datanodes was more meaningful for this image to make sense.  This way the default will give you a cluster-like environment by default. At least this was my perception when wanted to simulate a Spark cluster and actually see my task split.

File: docker-container/Dockerfile (hunk 3 above)
Comment (saifjsl) [reply docker-container/Dockerfile]: The original image does not exist anymore. Yet this is more bloated than the original. I would suggest testing and validating this one before merge. I didn't find any lightweight image like the original one.

I would also remove R dependencies or at least make them optional, since building R takes more than 2/3rds total build time. didn't commit that though.

**Comment by saifjsl** [comment]
Hi, no worries, I think this is a fantastic image for simulating cluster environments. It was just a bit outdated. And the base image does not exist anymore, although the replacement I proposed is not better either, and a bit more bloated.

File: cluster-setup/local-cluster/docker-compose.yml (hunk 1 above)
Comment (robvadai) [reply cluster-setup/local-cluster/docker-compose.yml]: Shouldn't this be `- "50570:50070"` ?
Because I can't see that you changed the port on the Hadoop side, ie. you just want to expose it to a different port on the host?

**Comment by robvadai** [comment]
@saifjsl hi there, sorry I didn't respond in the last few months, can you answer me the question regarding the port mapping, and can you also update the README please?

Then we can merge this change.

Thanks

File: cluster-setup/local-cluster/docker-compose.yml (hunk 1 above)
Comment (saifjsl) [reply cluster-setup/local-cluster/docker-compose.yml]: This branch would require serious updating. My idea here was to achieve two things overall:

1. Update the source image so we can re-build it (right now it is not possible, as the source does not exist anymore)

2. Update the ports so it works on Windows. But even I tried hard, I could not get it to work so I think port re-mapping is not important at all. We can wait until Windows WSL2 for better native linux docker support. Item 1 is definitely more important imho

**Comment by robvadai** [comment]
Hi there,

finally I had some time and updated all component versions.

Please see version 0.8.0 of the Docker image.

I hope it's OK, can we close this PR?

Sorry but I kept all config files as they were initially, just updated the component versions and now using the official OpenJDK base image.

**Comment by saif-ellafi** [comment]
Hi @robvadai glad to know you have some time. I also have some time available.

I want to test the upgraded version of the docker! As a second priority I was trying to make it work on Windows, but I can check that later on a different new branch.
